source/backend/dal/Repositories/ResearchFileRepository.cs

```diff
@@ -125,9 +130,11 @@ public PimsResearchFile Update(PimsResearchFile researchFile)
                 Context.Entry(deletedPurpose).State = EntityState.Deleted;
             }
 
-            researchFile.PimsResearchFilePurposes = purposes;
+            existingResearchFile.PimsResearchFilePurposes = purposes;
+
+            Context.Entry(existingResearchFile).CurrentValues.SetValues(researchFile);
```

Comment:
I prefer this method of saving updates, but it would be worth mentioning to Sue to run regression on research file update to make sure everything works as it did before.

Reply:
Good idea. I had to change the update method because the other way was not compatible with our "UpdateChild" method - was getting errors from EF